crates/autopilot/src/solvable_orders.rs

```diff
@@ -231,9 +232,23 @@ impl SolvableOrdersCache {
             prices,
         };
         let removed = counter.record(&auction.orders);
-        order_events.extend(removed.into_iter().map(|o| (o, OrderEventLabel::Filtered)));
-
-        self.database.store_order_events(&order_events).await;
```

Comment:
Do we still need the old method (I feel like all order events should only be updated if non-subsequent). If so, the method could also be called more neutral (ie, keep `store_order_events` but make it only store non subsequent events)

Reply:
It was already mentioned here: https://github.com/cowprotocol/services/issues/1977#issuecomment-1833830663. And I think it was additionally discussed with @MartinquaXD. Filtering other labels such as `Ready`, `Traded`, `Considered` extends the time period between the events. If that won't harm metrics/debugging, we can apply this logic to all the events.

Reply:
> Filtering other labels such as Ready, Considered extends the time period between the events.

Do you see any way in which this could make a difference? If I understand correctly `ready, considered, considered, traded` would become `ready, considered, traded` which seems fine. To me this metric should allow to track the  state machine for the lifecycle of an order.

Reply:
@MartinquaXD mentioned that the current metric where `order_evets` table is used is going to be improved. But I don't have enough context in which way.
Currently, we have something like `ready->ready->ready->traded` where we can clearly see the time difference between the latest `ready` and `traded` events. With filtering `ready` events, the difference becomes bigger. For some of the subsequent labels, we can only update their timestamps. But for the `invalid` label, for example, we care about the initial timestamp rather than the latest one. Anyways, would need more context from Martin 🙂

Reply:
I'm pretty sure we always care about the initial timestamp (for ready -> ready -> traded, we should measure our performance from when your order first became ready).

Given Martin is off until the 8th feel free to merge either with or without the unification (but please follow up before closing the issue)